src/main/java/org/sunbird/hubservices/daoimpl/GraphDao.java

```diff
@@ -25,292 +14,279 @@
-				if (!CollectionUtils.isEmpty(attributes)) {
-					attributes.stream().forEach(
-							attribute -> sb.append("n").append(level).append(".").append(attribute).append(","));
-					sb.deleteCharAt(sb.length() - 1);
-				} else {
-					sb.append("n").append(level);
-				}
-				query.append(sb).append(" Skip ").append(offset).append(" limit ").append(limit);
-
-				System.out.println("query-> " + query);
-				Statement statement = new Statement(query.toString(), parameters);
-
-				StatementResult result = transaction.run(statement);
-				List<Record> records = result.list();
-
-				transaction.commitAsync().toCompletableFuture();
-				logger.info("Neighbour users for UUID {} found successfully ", UUID);
-				return getNodes(records);
-
-			} catch (ClientException e) {
-				transaction.rollbackAsync().toCompletableFuture();
-				throw new GraphException(ErrorCode.GRAPH_TRANSACTIONAL_ERROR.name(), e.getMessage());
-
-			} finally {
-				transaction.close();
-			}
-		} catch (SessionExpiredException se) {
-			throw new GraphException(ErrorCode.GRAPH_SESSION_EXPIRED_ERROR.name(), se.getMessage());
-		} finally {
-			session.close();
-		}
-
-	}
+    private final Logger logger = LoggerFactory.getLogger(GraphDao.class);
+
+    @Autowired
+    private Driver neo4jDriver;
+
+    private final String label;
+
+    @Autowired
+    public GraphDao(String label) {
+        this.label = label;
+    }
+
+    @Override
+    public Boolean upsertNode(Node node) throws Exception {
+        Session session = neo4jDriver.session();
+        Transaction transaction = session.beginTransaction();
+        try {
+            Statement statement = new Statement("MATCH (n:" + label + ") WHERE n.id=$fromUUID " + "RETURN n", parameters("fromUUID", node.getId()));
+            StatementResult result = transaction.run(statement);
+            List<Record> existingNodes = result.list();
+            result.consume();
+            if (!existingNodes.isEmpty()) {
+                logger.info("Nodes exists, new node cannot be created! ");
+            } else {
+                System.out.println("Node doesn't exists, new node can be created! ");
+                Map<String, Object> params = new HashMap<>();
+                params.put(Constants.Graph.PROPS.getValue(), new ObjectMapper().convertValue(node, Map.class));
+                statement = new Statement("CREATE (n:" + label + ") SET n = $props RETURN n", params);
+                result = transaction.run(statement);
+                result.consume();
+                transaction.commitAsync().toCompletableFuture().get();
+                logger.info("user node with id {} created successfully ", node.getId());
+            }
+        } catch (Exception e) {
+            transaction.rollbackAsync().toCompletableFuture().get();
+            logger.error("user node creation failed : ", e);
+            return Boolean.FALSE;
+
+        } finally {
+            transaction.close();
+            session.close();
```

Comment:
these two statements should be inside another try catch... also need to check for != null before calling close.